src/Aqovia.DurableFunctions.Testing/TestJobHostWrapper.cs

```diff
@@ -145,24 +144,21 @@ public async Task WaitForOrchestrationAsync(string instanceId)
         {
             var timeout = Debugger.IsAttached ? TimeSpan.FromMinutes(5) : TimeSpan.FromSeconds(30);
 
-            _orchestrationState = (await _durabilityProvider.GetOrchestrationStateAsync(instanceId, allExecutions: false)).First();
-            _orchestrationState = await _durabilityProvider.WaitForOrchestrationAsync(instanceId, _orchestrationState.OrchestrationInstance.ExecutionId, timeout, CancellationToken.None);
+            var orchestrationState = (await _durabilityProvider.GetOrchestrationStateAsync(instanceId, allExecutions: false)).First();
+            await _durabilityProvider.WaitForOrchestrationAsync(instanceId, orchestrationState.OrchestrationInstance.ExecutionId, timeout, CancellationToken.None);
         }
 
         /// <summary>
-        /// GetLastOrchestrationStateWithHistoryAsync
-        /// Extracts the required orchestration state and the history of the orchestration
+        /// GetOrchestrationStateWithHistoryAsync
+        /// Extracts the required orchestration state and the history of the orchestration by instance Id
         /// </summary>
+        /// <param name="instanceId"></param>
         /// <returns>A tuple of required orchestration state and the history of the orchestration</returns>
-        public async Task<(OrchestrationState, string)> GetLastOrchestrationStateWithHistoryAsync()
+        public async Task<(OrchestrationState, string)> GetOrchestrationStateWithHistoryAsync(string instanceId)
         {
-            if(_orchestrationState == null)
-            {
-                throw new Exception("Orchestration state is null");
-            }
-
-            var history = await _durabilityProvider.GetOrchestrationHistoryAsync(_orchestrationState.OrchestrationInstance.InstanceId, _orchestrationState.OrchestrationInstance.ExecutionId);
-            return (_orchestrationState, history);
+            var orchestrationState = (await _durabilityProvider.GetOrchestrationStateAsync(instanceId, allExecutions: false)).First();
+            var history = await _durabilityProvider.GetOrchestrationHistoryAsync(orchestrationState.OrchestrationInstance.InstanceId, orchestrationState.OrchestrationInstance.ExecutionId);
```

Comment:
Is it safe to not check for null?
There was a check in the previous function.
If state is null, the getOrchestrationHistoryAsync will try to access a valu on that null state?

Reply:
Suggest leaving this until it is an issue - have not observed this problem as yet - will patch if required